src/libcore/option.rs

```diff
@@ -972,6 +972,92 @@ impl<T: Clone> Option<&mut T> {
     }
 }
 
+impl<T: fmt::Debug> Option<T> {
```

Comment:
Hmm... this makes it rather less useful... :(

Another case for https://github.com/rust-lang/rust/pull/49071?

Reply:
It is consistent with all of the other unwrap/expect methods though.

Reply:
Which ones do you mean? (This compiles fine despite no `T: Debug` constraint
```rust
fn foo<T>(x: Option<T>) -> T { x.unwrap() }
```
)

Reply:
To be clear I'd rather land this with the constraint than not land it at all; I'm just a bit bummed is all. =)

Reply:
I mean the "bad" variant is debug-printed, at least when it has a value. For `Result<T, E>`, `unwrap()` requires `E: Debug` and `unwrap_err()` requires `T: Debug`. There's nothing to print for `Option::unwrap()` when it encounters `None`, so that's universal anyway, but this new `unwrap_none()` is akin to `unwrap_err()` and requires `T: Debug` too.

Reply:
Ah; good point 👍